File: app/src/main/java/com/katkov/lolachievements/data/cloud/repository/SummonerInfoRepository.java
package com.katkov.lolachievements.data.cloud.repository;

import com.katkov.lolachievements.data.cloud.api.ApiService;
import com.katkov.lolachievements.data.mapper.SummonerDTOApiToDomainMapper;
import com.katkov.lolachievements.domain.model.SummonerDTO;

import javax.inject.Inject;

import io.reactivex.Single;
import io.reactivex.schedulers.Schedulers;

public class SummonerInfoRepository {

    private final ApiService apiService;
    private final SummonerDTOApiToDomainMapper mapper;
    private final String apiKey = "RGAPI-70b77dc5-fc20-4051-b3cb-4ead302e5f9f";

    @Inject
    public SummonerInfoRepository(ApiService apiService, SummonerDTOApiToDomainMapper mapper) {
        this.apiService = apiService;
        this.mapper = mapper;
    }

    public Single<SummonerDTO> getSummonerDTO(String summonerName) {
        return apiService.getSummonerDTO(summonerName, apiKey)
                .map(apiModel -> mapper.map(apiModel))
                .subscribeOn(Schedulers.io());
    }
}

File: app/src/main/java/com/katkov/lolachievements/data/cloud/repository/SummonerInfoRepository.kt
package com.katkov.lolachievements.data.cloud.repository

import com.katkov.lolachievements.data.cloud.api.ApiService
import com.katkov.lolachievements.data.cloud.utils.ApiUtils.API_KEY
import com.katkov.lolachievements.data.local.prefser.LoginModelHolder
import com.katkov.lolachievements.data.mappers.SummonerApiDtoToDomainMapper
import com.katkov.lolachievements.domain.model.SummonerDto
import io.reactivex.Single
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject

class SummonerInfoRepository
@Inject
constructor(
    private val apiService: ApiService,
    private val mapperDto: SummonerApiDtoToDomainMapper,
    private val loginModelHolder: LoginModelHolder
) {

    fun getSummonerDTO(): Single<SummonerDto> {
        val summonerName = loginModelHolder.getLoginModel()!!.summonerName

        return apiService.getSummonerApiDto(summonerName, API_KEY)
            .map { mapperDto.map(it) }
            .subscribeOn(Schedulers.io())
    }
}
